datanommer.models/datanommer/models/view.py

```diff
@@ -0,0 +1,43 @@
+from sqlalchemy import DDL, event, func, inspect, select, text
+from sqlalchemy.ext import compiler
+from sqlalchemy.schema import DDLElement
+
+
+class CreateMaterializedView(DDLElement):
+    def __init__(self, name, selectable):
+        self.name = name
+        self.selectable = selectable
+
+
+@compiler.compiles(CreateMaterializedView)
+def _create_view(element, compiler, **kw):
+    selectable = compiler.sql_compiler.process(element.selectable, literal_binds=True)
+    return f"CREATE MATERIALIZED VIEW IF NOT EXISTS {element.name} AS {selectable}"
+
+
+def view_exists(ddl, target, connection, **kw):
+    return ddl.name in inspect(connection).get_view_names()
+
+
+def view_doesnt_exist(ddl, target, connection, **kw):
+    return not view_exists(ddl, target, connection, **kw)
+
+
+def create_view():
+    from . import DeclarativeBase, Message
+
+    selectable = (
+        select(Message.topic, func.count().label("message_count"))
+        .where(Message.timestamp >= text("NOW() - INTERVAL '1 year'"))
```

Comment:
I would like to make the interval a module-level variable, so that we can change it easily if necessary.